types/src/v0/impls/reward.rs

```diff
@@ -413,19 +413,22 @@ pub async fn first_two_epochs(height: u64, instance_state: &NodeState) -> anyhow
     Ok(epoch == first_epoch || epoch == first_epoch + 1 || epoch == EpochNumber::new(0))
 }
 
-pub async fn catchup_missing_accounts(
+pub async fn find_validator_info(
     instance_state: &NodeState,
     validated_state: &mut ValidatedState,
     parent_leaf: &Leaf2,
     view: ViewNumber,
+    height: u64,
 ) -> anyhow::Result<Validator<BLSPubKey>> {
     let parent_height = parent_leaf.height();
-    let new_height = parent_height + 1;
-    let epoch_height = instance_state
-        .epoch_height
-        .context("epoch height not found")?;
-    // This block is for the next block we need the leader for that block not the parent block
-    let epoch = EpochNumber::new(epoch_from_block_number(new_height, epoch_height));
+    let parent_view = parent_leaf.view_number();
+
+    let epoch_height = instance_state.epoch_height.context("epoch height")?;
+    if epoch_height == 0 {
+        bail!("epoch height is 0. can not catchup reward accounts");
+    }
+    let epoch = EpochNumber::new(epoch_from_block_number(height, epoch_height));
+
```

Comment:
I'd rather not include these changes, to me it just makes it less clear to read with the height being passed in.  It's not clear from the fn signature which `height`

Reply:
reverted